components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -383,38 +383,49 @@ void EthTxController::ApproveHardwareTransaction(
   }
 }
 
+void EthTxController::GetTransactionMessageToSign(
+    const std::string& tx_meta_id,
+    GetTransactionMessageToSignCallback callback) {
+  std::unique_ptr<EthTxStateManager::TxMeta> meta =
+      tx_state_manager_->GetTx(tx_meta_id);
+  if (!meta) {
+    LOG(ERROR) << "No transaction found";
+    std::move(callback).Run("");
```

Comment:
returning an empty string is not great imo and we should avoid this pattern. It's better to return an optional value and some kind of error message so the caller knows why it failed.

Reply:
changed to use optional type